### PR TITLE
Add support for importing from VMDK, VDI, VHD, VHDX formats

### PR DESCRIPTION
**Pull request by maya-r**
**What this PR does / why we need it**:
Sometimes images are only available in these formats as file, e.g. Microsoft only offers these formats for Windows VMs.
qemu-img supports them already, we just need to allow it & test it.
It's also a lot simpler to export a file and import it than a warm import from VMware.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Closes #1512

**Special notes for your reviewer**:

**Release note**:

```release-note
Add support for importing from VMDK, VDI, VHD, VHDX formats
```

### CODE REVIEW

File: pkg/image/filefmt_test.go [hunk 1]
```diff
@@ -60,6 +60,14 @@ var _ = Describe("File format tests", func() {
 			fields{"gz", []byte{0x1F, 0x8B}, 0, 0, 0},
 			[]byte{'Q', 'F', 'I', 0xfb},
 			false),
+		table.Entry("match vmdk",
```
Comment (arnongilboa) [inline pkg/image/filefmt_test.go:63]: what about vhd & vhdx?

File: pkg/image/qemu.go [hunk 2]
```diff
@@ -195,7 +195,7 @@ func (o *qemuOperations) Info(url *url.URL) (*ImgInfo, error) {
 
 func isSupportedFormat(value string) bool {
 	switch value {
-	case "raw", "qcow2":
+	case "raw", "qcow2", "vmdk", "vdi", "vpc", "vhdx":
```
Comment (arnongilboa) [inline pkg/image/qemu.go:198]: shouldn't  "vpc" be "vhd"?

File: tests/datavolume_test.go [hunk 3]
```diff
@@ -572,6 +584,102 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 					Message: "Import Complete",
 					Reason:  "Completed",
 				}}),
+			table.Entry("succeed creating import dv with given valid vmdk url (https)", dataVolumeTestArguments{
```
Comment (arnongilboa) [inline tests/datavolume_test.go:587]: all similar, smells like a refactor needed

File: tests/utils/fileConversion.go [hunk 4]
```diff
@@ -141,6 +145,50 @@ func toQcow2(srcfile, tgtDir string) (string, error) {
 	return tgt, nil
 }
 
+func toVmdk(srcfile, tgtDir string) (string, error) {
```
Comment (arnongilboa) [inline tests/utils/fileConversion.go:148]: again, all similar, smells like a refactor needed

**Review by arnongilboa — CHANGES_REQUESTED** [review]

File: pkg/image/qemu.go (hunk 2 above)
Comment (maya-r) [reply pkg/image/qemu.go:198]: nope qemu-img strangely calls it vpc

Comment (arnongilboa) [reply pkg/image/qemu.go:198]: ok :)

File: pkg/image/filefmt_test.go (hunk 1 above)
Comment (maya-r) [reply pkg/image/filefmt_test.go:63]: ah didn't notice that the original pr added vmdk and vdi and didn't add my new vhd and vhdx here

File: tests/datavolume_test.go [hunk 5]
```diff
@@ -1020,6 +924,27 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		)
 	})
 
+	table.DescribeTable("Succeed HTTPS import in various formats", func(url func() string) {
```
Comment (arnongilboa) [inline tests/datavolume_test.go:927]: Looks much better now! :)

File: pkg/image/filefmt_test.go (hunk 1 above)
Comment (maya-r) [reply pkg/image/filefmt_test.go:48]: Added those

File: tests/utils/fileConversion.go (hunk 4 above)
Comment (tomob) [reply tests/utils/fileConversion.go:148]: +1

Comment (maya-r) [reply tests/utils/fileConversion.go]: done

File: pkg/image/filefmt.go [hunk 6]
```diff
@@ -46,6 +46,30 @@ var knownHeaders = Headers{
 		SizeOff: 0,
 		SizeLen: 0,
 	},
+	"vmdk": Header{
+		Format:      "vmdk",
+		magicNumber: []byte{0x4B, 0x44, 0x4D, 0x56},
```
Comment (arnongilboa) [inline pkg/image/filefmt.go:51]: Why not have consts for the magic numbers? each of them appears 3 times.

**Review by arnongilboa — CHANGES_REQUESTED** [review]

File: pkg/image/filefmt.go (hunk 6 above)
Comment (arnongilboa) [reply pkg/image/filefmt.go:51]: here are some ideas:
https://qvault.io/2019/10/21/golang-constant-maps-slices/

Comment (arnongilboa) [reply pkg/image/filefmt.go:51]: @maya-r decided not to fix this one?

Comment (maya-r) [reply pkg/image/filefmt.go]: I did something a bit different and used meaningful strings

Comment (arnongilboa) [reply pkg/image/filefmt.go]: nice! but having every meaningful string x 3 is a must?

**Comment by maya-r** [comment]
ping @mhenriks @awels @aglitke

**Review by arnongilboa — APPROVED** [review]

**Pull request merged**